src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -56,13 +53,12 @@ protected override bool Matches(string actual)
         }
 
         /// <summary>
-        /// Flag the constraint to use the supplied StringComparison type.
+        /// Overrides the <see cref="StringConstraint.IgnoreCase"/> setting and instead uses the specified comparison.
         /// </summary>
-        /// <param name="comparisonType ">The StringComparison type to be used.</param>
-        /// <returns>Self.</returns>
-        public SubstringConstraint Using(StringComparison comparisonType)
+        public SubstringConstraint Using(StringComparison? comparisonType)
         {
-            this.comparisonType = comparisonType;
+            this.comparisonType = comparisonType ?? (caseInsensitive ? 
+                StringComparison.CurrentCultureIgnoreCase : StringComparison.Ordinal);
```

Comment:
Just a suggestion: Why don't we let `indexOf` decides which `StringComparison` to use. I mean we keep `SubstringConstraint.comparisonType` nullable also and change `Matches` method instead to be
 `protected override bool Matches(string actual)
  {
        if (actual == null)
              return false;

        if (comparisonType == null)
             return actual.IndexOf(expected) >= 0;

       return actual.IndexOf(expected, comparisonType) >= 0;
  }`